### PR TITLE
fix remaining analyzer deprecation warnings

### PR DESCRIPTION
**Pull request by jakemac53**
Some places I did change the code to utilize the new methods better and add some type checks, others I just added an explicit cast.

### CODE REVIEW

File: build_resolvers/lib/src/resolver.dart [hunk 1]
```diff
@@ -152,9 +151,9 @@ class AnalyzerResolver implements ReleasableResolver {
   @override
   Future<bool> isLibrary(AssetId assetId) async {
     var source = _driver.sourceFactory.forUri2(assetId.uri);
-    return source != null &&
-        source.exists() &&
-        (await _driver.getSourceKind(assetPath(assetId))) == SourceKind.LIBRARY;
+    if (source == null || !source.exists()) return false;
+    var result = _driver.getFileSync2(assetPath(assetId));
+    return result is FileResult && !result.isPart;
```
Comment (natebosch) [inline build_resolvers/lib/src/resolver.dart:156]: one thing I was a little worried about is changing behavior subtly in cases where we get an invalid result.

If we introduce a bug down the line, will we start treating some libraries and non-libraries when the real situation is using the analyzer incorrectly?

**Review by natebosch — APPROVED** [review]

**Comment by jakemac53** [comment]
It looks like this did also change the behavior in some cases. I will take a look at those tests.

**Comment by jakemac53** [comment]
I went ahead and changed the logic more back to how it was, it looks like it was important for us to explicitly re-parse the file or we get stale results.

**Review by natebosch — APPROVED** [review]

**Comment by jakemac53** [comment]
merging this and following up with a bug fix for the other issue showing up.

File: build_resolvers/lib/src/resolver.dart (hunk 1 above)
Comment (natebosch) [reply build_resolvers/lib/src/resolver.dart]: Do we already know that non-Dart files won't show up here? Could a non-part, non-library file show up as if it was a library?

cc @scheglov - this is similar to the internal code so the same concerns likely apply.

**Pull request merged**